### PR TITLE
Bump proof-systems for refactors

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR builds upon #12325, completing the update of the `proof-systems` submodule for the PR chain ending in https://github.com/o1-labs/proof-systems/pull/839.

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/crypto/kimchi_bindings/stubs/src/plonk_verifier_index.rs [hunk 1]
```diff
@@ -70,13 +70,60 @@ where
     }
 }
 
+#[derive(ocaml::IntoValue, ocaml::FromValue, ocaml_gen::Struct)]
+pub struct CamlLookupInfo {
```
Comment (mimoo) [inline src/lib/crypto/kimchi_bindings/stubs/src/plonk_verifier_index.rs:74]: why not use LookupInfo directly? I'm guessing the only things that didn't work out were the `usize`? In which case we can just support them in ocaml-gen or use u32 instead?

**Review by mimoo — APPROVED** [review]

File: src/lib/crypto/kimchi_bindings/stubs/src/plonk_verifier_index.rs (hunk 1 above)
Comment (mrmr1993) [reply src/lib/crypto/kimchi_bindings/stubs/src/plonk_verifier_index.rs:74]: > also all of these types are usually defined in the proof-system crate

I think you started doing that, but there are still several that live here. We should migrate them over, but I'd rather not do it in this PR.

Comment (mimoo) [reply src/lib/crypto/kimchi_bindings/stubs/src/plonk_verifier_index.rs:74]: > I think you started doing that

this predates me o.o

**Pull request merged**